modules/sdk-coin-avaxp/src/lib/exportTxBuilder.ts

```diff
@@ -0,0 +1,128 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import { BuildTransactionError, NotSupported, TransactionType } from '@bitgo/sdk-core';
+import { TransactionBuilder } from './transactionBuilder';
+import {
+  BaseTx,
+  ExportTx,
+  PlatformVMConstants,
+  SECPTransferOutput,
+  TransferableOutput,
+  Tx,
+  UnsignedTx,
+} from 'avalanche/dist/apis/platformvm';
+import utils from './utils';
+import { BN, Buffer as BufferAvax } from 'avalanche';
+import { AmountOutput } from 'avalanche/dist/apis/evm/outputs';
+
+export class ExportTxBuilder extends TransactionBuilder {
+  private _amount: BN;
+  private _targetChainId: BufferAvax;
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+    this.transaction._fee = new BN(this.transaction._network.txFee);
+  }
+
+  protected get transactionType(): TransactionType {
+    return TransactionType.Export;
+  }
+
+  /**
+   *
+   * @param value
+   */
+  targetChainId(chainId: string | Buffer): this {
+    const newTargetChainId = typeof chainId === 'string' ? utils.cb58Decode(chainId) : BufferAvax.from(chainId);
+    this.validateChainId(newTargetChainId);
+    this._targetChainId = newTargetChainId;
+    return this;
+  }
+
+  /**
+   *
+   * @param value
+   */
+  amount(value: BN | string): this {
+    const valueBN = BN.isBN(value) ? value : new BN(value);
+    this.validateAmount(valueBN);
+    this._amount = valueBN;
+    return this;
+  }
+
+  /**
+   *
+   * @param amount
+   */
+  validateAmount(amount: BN): void {
+    if (amount.lten(0)) {
+      throw new BuildTransactionError('Amount must be greater than 0');
+    }
+  }
+
+  /**
+   *
+   * @param chainID
+   */
+  private validateChainId(chainID: BufferAvax) {
+    if (chainID.length != 32) {
+      throw new BuildTransactionError('Chain id are 32 byte size');
+    }
+  }
+
+  initBuilder(tx: Tx): this {
+    super.initBuilder(tx);
+    const baseTx: BaseTx = tx.getUnsignedTx().getTransaction();
+    if (!this.verifyTxType(baseTx)) {
+      throw new NotSupported('Transaction cannot be parsed or has an unsupported transaction type');
+    }
+    this._targetChainId = baseTx.getDestinationChain();
+    this._amount = (baseTx.getExportOutputs()[0].getOutput() as any as AmountOutput).getAmount();
```

Comment:
why do we only take the first output's amount?

Reply:
Ok, I could review full array, but I should be only one, if it's builded by SDK

Reply:
If we have more than 1 output we should either count them or throw an exception if we only expect 1